addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -218,6 +218,19 @@ def _send_sms_get_composer_values(self, res_ids):
             'mass_sms_allow_unsubscribe': self.sms_allow_unsubscribe,
         }
 
+    def action_send_skip_queue(self, res_ids=None):
+        mass_sms = self.filtered(lambda m: m.mailing_type == 'sms')
+        if mass_sms:
+            if len(mass_sms._get_remaining_recipients()) > 0:
+                mass_sms.action_send_sms(res_ids=res_ids)
+            else:
+                mass_sms.write({
+                    'state': 'done',
+                    'sent_date': fields.Datetime.now(),
+                    'kpi_mail_required': not mass_sms.sent_date,
```

Comment:
you don't need to create a new method action_send_skip_queue.
modify directly action_send_mail.
also, you cannot write on mass_sms using mass.sent_date as mass_sms could contain multiple sms (and various sent_date)

Reply:
I created this new method because the method action_send_mail can be called by the cron task which will apply the method :

`_process_mass_mailing_queue`

from odoo/addons/mass_mailing/models/mailing.py

and this method is performing similar actions to what I want to achieve in 

`action_send_skip_queue`

and the write on multiple records is intentional as it handle the case when the domain doesn't match with any record, and as such no message will be sent at all.

Reply:
I think this could be done by only doing this

```
def action_send_sms(self, res_ids=None):
    for mailing in self:
        if not res_ids:
            res_ids = mailing._get_remaining_recipients()
        if res_ids:
            composer = self.env['sms.composer'].with_context(active_id=False).create(mailing._send_sms_get_composer_values(res_ids))
            composer._action_send_sms()
        mailing.write({
            'state': 'done',
            'sent_date': fields.Datetime.now(),
            'kpi_mail_required': not mailing.sent_date,
            })
    return True
```